docs/content/dagster-cloud/developing-testing/dagster-cloud-yaml.mdx

```diff
@@ -0,0 +1,169 @@
+---
+title: Confiuguration options for dagster_cloud.yaml | Dagster Docs
+description: How to use dagster_cloud.yaml to configure Dagster Cloud
+---
+
+# Dagster_cloud.yaml Guide
+
+<Note>This guide is applicable to Dagster Cloud.</Note>
+
+This dagster_cloud.yaml file is used to configure a Dagster Cloud instance. The file is required to tell Dagster where the code sources are to load into Dagster Cloud but has other options as well. For Hybrid Deployments, secrets are configured in this file as well.
+
+## Where to put the dagster_cloud.yaml file
+
+The dagster_cloud.yaml file should be placed in your dagster project's repository root. Below is an example of a file structure from the [Dagster cloud ETL quickstart](https://github.com/dagster-io/dagster/tree/master/examples/quickstart_etl).
+
+```shell
+quickstart_etl
+├── README.md
+├── quickstart_etl
+│   ├── __init__.py
+│   ├── assets
+├── quickstart_etl_tests
+├── dagster_cloud.yaml
+├── pyproject.toml
+├── setup.cfg
+└── setup.py
+```
+
+## Code Source
+
+The only **mandatory** section in the dagster_cloud.yaml file is setting up your code locations. If you have multiple code locations, you will have multiple code locations within a single dagster_cloud.yaml file. Dagster code can be sourced from a package, a python file, or a module.
+
+Parameter reference table:
+
+| Parameter                            | Description                                                                           | Data type            |
+| ------------------------------------ | ------------------------------------------------------------------------------------- | -------------------- |
+| `locations.location_name`            | The name of your code location that will appear in the Dagster UI code locations page | string               |
+| `locations.code_source.package_name` | The folder that contains your dagster code                                            | string (folder name) |
+| `locations.code_source.python_file`  | The python file that contains your dagster code                                       | string (.py file)    |
+| `locations.code_source.module_name`  | The module that contains your dagster code                                            | string (module name) |
+
+Single code location:
+
+```yaml
+#dagster_cloud.yaml
+
+locations:
+  - location_name: example_location
+    code_source:
+      package_name: my_dagster_project
+    build:
+      directory: ./
+      registry: 764506304434.dkr.ecr.us-west-2.amazonaws.com/branch-deployments-gh-action-test
+```
+
+Multiple code locations:
+
+```yaml
+#dagster_cloud.yaml
+
+locations:
+  - location_name: data-eng-pipeline
+    code_source:
+      package_name: hooli_data_eng
+  - location_name: basics
+    code_source:
+      python_file: definitions.py
```

Comment:
the example here is a little odd, i think typically if you had multiple code locations in a single repo they would each be in a separate subfolder. Maybe if the python_file was like `hool_ml_pipeline/definitions.py`?